app/models/service/aggregation.rb

```diff
@@ -42,26 +273,51 @@ def aggregate_direct_vm_memory_on_disk
   end
 
   def aggregate_all_vm_cpus
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.cpu_total_cores.to_i }
+    if has_attribute?("aggregate_all_vm_cpus")
+      self["aggregate_all_vm_cpus"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.cpu_total_cores.to_i }
+    end
   end
 
   def aggregate_all_vm_memory
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.ram_size.to_i }
+    if has_attribute?("aggregate_all_vm_memory")
+      self["aggregate_all_vm_memory"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.ram_size.to_i }
+    end
   end
 
   def aggregate_all_vm_disk_count
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.num_disks.to_i }
+    if has_attribute?("aggregate_all_vm_disk_count")
+      self["aggregate_all_vm_disk_count"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.num_disks.to_i }
+    end
   end
 
   def aggregate_all_vm_disk_space_allocated
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.allocated_disk_storage.to_i }
+    if has_attribute?("aggregate_all_vm_disk_space_allocated")
+      self["aggregate_all_vm_disk_space_allocated"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.allocated_disk_storage.to_i }
+    end
   end
 
   def aggregate_all_vm_disk_space_used
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.used_disk_storage.to_i }
+    if has_attribute?("aggregate_all_vm_disk_space_used")
+      self["aggregate_all_vm_disk_space_used"]
+    else
+      all_vms.inject(0) { |aggregate, vm| aggregate + vm.used_disk_storage.to_i }
+    end
   end
 
   def aggregate_all_vm_memory_on_disk
-    all_vms.inject(0) { |aggregate, vm| aggregate + vm.ram_size_in_bytes_by_state.to_i }
+    if has_attribute?("aggregate_all_vm_memory_on_disk")
+      # Avoids (poorly) "ERROR:  integer out of range" in postgres
+      self["aggregate_all_vm_memory_on_disk"] * 1.megabyte
```

Comment:
I want to say it wasn't as simple as doing that, and it was easier to do that one multiplication is ruby instead of trying to cast it in SQL (again, memory failing me here).

That said, it may have been as simple as "hmm... maybe I should make this as simple as I can for @Fryguy to approve this and not add more Arel...", which adding another cast would do.  Not really opinionated one way or the other at this point, but would need to re-test things if you want the change.